lib/JLArrays/src/JLArrays.jl

```diff
@@ -387,4 +395,12 @@ Adapt.adapt_storage(::JLBackend, a::Array) = Adapt.adapt(JLArrays.JLArray, a)
 Adapt.adapt_storage(::JLBackend, a::JLArrays.JLArray) = a
 Adapt.adapt_storage(::KernelAbstractions.CPU, a::JLArrays.JLArray) = convert(Array, a)
 
+# Caching Allocator.
+
+const JLACacheAllocator = GPUArrays.AllocCache.PerDeviceCacheAllocator(JLArray)
+
+GPUArrays.AllocCache.cache_allocator(::Type{<:JLArray}) = JLACacheAllocator
```

Comment:
Why is this needed, now that you switched to the array type? Isn't all information there for the caller to construct an appropriate allocator cache?

Reply:
This is for the internal implementation to retrieve the actual cache for `@enable`.
E.g. when CUDA calls [`alloc!`](https://github.com/JuliaGPU/GPUArrays.jl/pull/576/files#diff-fcdd73df7e8055bf03e5d08337fcd21346ddb4d4f4f6de2cf82e9979d1e1df7fR103) we retrieve its allocator cache based on its array type.
Otherwise the user would have to pass the cache itself to the macro, no?

Reply:
I mean that we can get rid of the alias, and replace calls to `cache_allocator` by `AllocCache.PerDeviceCacheAllocator(AT)`. Just trying to minimize the interface to be implemented by back-ends.

Reply:
I'm not opposed, just a bit confused.
`AllocCache.PerDeviceCacheAllocator(AT)` is call to ctor, when with `cache_allocator` we retrieve an instance.
In CUDA.jl we [define a global variable](https://github.com/JuliaGPU/CUDA.jl/pull/2593/files#diff-1177a809b2ffc09e77e53c0c8a036ca503494f88c39c10b6027542670d444b0aR1) that we then retrieve with `cache_allocator`.
How are we going to access this variable with `AllocCache.PerDeviceCacheAllocator(AT)` if its a call to ctor?

Or is the point just to rename `cache_allocator` to `AllocCache.PerDeviceCacheAllocator`? But then it is ambiguous because ctor has the same method signature.